### ISSUE TITLE
ListView doesn't show items when navigating and using a promise

### ISSUE THREAD

**Issue by risp** [winjs/angular-winjs#28]
Repro page
```html
<!DOCTYPE html>
<html ng-app="testApp">
<head>
    <link rel="stylesheet" href="lib/Microsoft.WinJS.4.0/css/ui-light.css"/>
    <script src="lib/angular.js"></script>
    <script src="lib/angular-route.js"></script>
    <script src="lib/Microsoft.WinJS.4.0/js/WinJS.min.js"></script>
    <script src="lib/angular-winjs.js"></script>
    <link rel="stylesheet" href="css/default.css"/>
</head>
<body>

<script>
    var app = angular.module('testApp', ['ngRoute', 'winjs']);

    app.config(['$routeProvider',
        function ($routeProvider) {
            $routeProvider.
                    when('/list', {
                        template: '<win-list-view item-data-source="data"><win-item-template><a href="#/detail">{{item.data}}</a></win-item-template></win-list-view>Regular repeat:<br/><div ng-repeat="item in data">{{item}}</div>',
                        controller: 'ListCtrl'
                    }).
                    when('/detail', {
                        template: '<a href="#/list">Back</a>',
                        controller: function() {}
                    }).
                    otherwise({
                        redirectTo: '/list'
                    });
        }]);

    app.controller('ListCtrl', ['$scope', 'DataService',
        function ($scope, DataService) {
            DataService.getData().then(function (data) {
                $scope.data = data;
            });
        }
    ]);

    app.factory('DataService', ['$http', function ($http) {
        var promise;

        return {
            getData: function () {
                if(!promise) {
                    promise = $http.get('https://www.google.com').then(function (data) {
                        return ["A", "B", "C"];
                    });
                }
                return promise;
            }
        };
    }
    ]);


    WinJS.UI.processAll();
</script>


<div class="view-container">
    <div ng-view class="view-frame"></div>
</div>

</body>
</html>
```
Repro steps:
1. Open page
2. Click on the first item 'A'
3. Now click the back link on top of the new page

Expected: The same 2 list as in step 1
Actual: The first WinJS list is not shown, the regular ng-repeat is still showing data (as expected)

This only seems to happen when using promises to return the data.

**Comment by risp** [comment]
Any idea when this issue will be fixed?

**Comment by TheWrathOfConst** [comment]
The fix for this is now in code review. Should be in master sometime early next week

**Comment by TheWrathOfConst** [comment]
Should now be fixed by commit 6c272d1

**Comment by risp** [comment]
Thanks a lot!
